modules/oe_theme_content_publication/oe_theme_content_publication.module

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * @file
+ * Module file used for theming the Publication content type.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Component\Utility\NestedArray;
+
+/**
+ * Implements template_preprocess_node() for the Publication node type.
+ */
+function oe_theme_content_news_preprocess_node__oe_publication__full(&$variables) {
+  if (NestedArray::keyExists($variables, ['content', 'oe_documents'])) {
```

Comment:
Since:

1. We are renaming the library to be more generic and
2. It's configured that the full view mode of the publication will show the documents,

This `if` conditional is not needed. You can just attach the library to the top level `$variables`.